Objects/Converters/ConverterAutocadCivil/ConverterAutocadCivilShared/ConverterAutocadCivil.Civil.cs

```diff
@@ -1128,13 +1122,155 @@ public Pipe PipeToSpeckle(PressurePipe pipe)
 
   // corridors
   // this is composed of assemblies, alignments, and profiles, use point codes to generate featurelines (which will have the 3d curve)
+
+  private Base AppliedSubassemblyParamToSpeckle(IAppliedSubassemblyParam param)
+  {
+    Base baseParam = new() { 
+      ["name"] = param.DisplayName,
+      ["key"] = param.KeyName, 
+      ["type"] = param.ValueType.Name, 
+      ["value"] = param.ValueAsObject,
+      ["designValue"] = param.DesignValueAsObject};
+
+    return baseParam;
+  }
+
+  private CivilAppliedSubassembly AppliedSubassemblyToSpeckle(AppliedSubassembly appliedSubassembly)
+  {
+    // retrieve subassembly name
+    Subassembly subassembly = Trans.GetObject(appliedSubassembly.SubassemblyId, OpenMode.ForRead) as Subassembly;
+
+    // get the calculated shapes
+    List<CivilCalculatedShape> speckleShapes = new();
+    foreach (CalculatedShape shape in appliedSubassembly.Shapes)
+    {
+      CivilCalculatedShape speckleShape = CalculatedShapeToSpeckle(shape);
+      speckleShapes.Add(speckleShape);
+    }
+
+    List<Base> speckleParameters = appliedSubassembly.Parameters.Select(p => AppliedSubassemblyParamToSpeckle(p)).ToList();
+
+    CivilAppliedSubassembly speckleAppliedSubassembly = new(appliedSubassembly.SubassemblyId.ToString(), subassembly.Name, speckleShapes, speckleParameters);
+    return speckleAppliedSubassembly;
+  }
+
+  private CivilAppliedAssembly AppliedAssemblyToSpeckle(AppliedAssembly appliedAssembly)
+  {
+    // get the applied subassemblies
+    List<CivilAppliedSubassembly> speckleSubassemblies = new();
+    foreach (AppliedSubassembly appliedSubassembly in appliedAssembly.GetAppliedSubassemblies())
+    {
+      CivilAppliedSubassembly speckleSubassembly = AppliedSubassemblyToSpeckle(appliedSubassembly);
+      speckleSubassemblies.Add(speckleSubassembly);
+    }
+
+    CivilAppliedAssembly speckleAppliedAssembly = new(speckleSubassemblies, appliedAssembly.AdjustedElevation, ModelUnits);
+    return speckleAppliedAssembly;
+  }
+
+  private CivilBaselineRegion BaselineRegionToSpeckle(BaselineRegion region)
+  {
+    // get the region assembly
+    Assembly assembly = Trans.GetObject(region.AssemblyId, OpenMode.ForRead) as Assembly;
+
+    // get the applied assemblies
+    List<CivilAppliedAssembly> speckleAppliedAssemblies = new();
+    foreach (AppliedAssembly appliedAssembly in region.AppliedAssemblies)
```

Comment:
If possible, it would be helpful within the AppliedAssembly object to have the station at which it is applied.
The Civil API doesnt give you this directly, but if you use a for loop to loop through the SortedStations, you can get the station for each AppliedAssembly, You then just need to pass that in when creating the Speckle AppliedAssembly

```
for (int i = 0; i < sortedStations.Length; i++)
{
    var station = sortedStations[i];

    var appliedAssembly= region.AppliedAssemblies[i];

   CivilAppliedAssembly speckleAssembly = AppliedAssemblyToSpeckle(appliedAssembly, station);

   speckleAppliedAssemblies.Add(speckleAssembly);
}
```

Reply:
I'll add the station property dynamically, since it's not accessible on the `appliedAssembly`